src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/MaxLengthAttribute.cs

```diff
@@ -87,9 +87,7 @@ public override bool IsValid(object value)
                 }
                 else
                 {
-                    // A cast exception previously occurred if a non-{string|array} property was passed
-                    // in so preserve this behavior if the value does not implement ICollection
-                    length = ((Array)value).Length;
+                    throw new InvalidCastException(SR.Format(SR.MaxLengthAttribute_InvalidValueType, value.GetType()));
```

Comment:
I think this was addressing [https://github.com/dotnet/corefx/issues/18361](https://github.com/dotnet/corefx/issues/18361)

---
In reply to: [121792346](https://github.com/dotnet/corefx/pull/20934#discussion_r121792346) [](ancestors = 121792346,121382258)